app/models/miq_filter.rb

```diff
@@ -130,7 +130,7 @@ def self.belongsto2object_list(tag)
 
       obj.with_relationship_type('ems_metadata') do
         obj.children.each do |c|
-          if c.class.to_s == klass && c.name == name
+          if c.kind_of?(klass.constantize) && c.name == name
```

Comment:
It appears that `klass` is the actual class in other parts of this file. Can you do `klass = klass.constantize` on line 119 and change 128 to be renamed to avoid re-using the same variable name with two different meanings.

```
tag_part_klass, name = p.split("|")
tag_part_klass = tag_part_klass.constantize
```

So this line, 133 becomes: `if c.kind_of?(tag_part_klass) && c.name == name`